src/Materials/Textures/imageCubeTexture.ts

```diff
@@ -0,0 +1,194 @@
+import { PanoramaToCubeMapTools } from '../../Misc/HighDynamicRange/panoramaToCubemap';
+import { Engine } from "../../Engines/engine";
+import { BaseTexture } from '../../Materials/Textures/baseTexture';
+import { Texture } from '../../Materials/Textures/texture';
+import { Scene } from "../../scene";
+import { Nullable } from "../../types";
+
+/**
+ * This represents a texture coming from an image supported by the web browser canvas.
+ */
+export class ImageCubeTexture extends BaseTexture {

```

Comment:
The name is a bit misleading perhaps? All cube textures are based on image.
We may need to find a name conveying the idea that it is only one 2d equirectangular image :)

Reply:
When I started I was using `JPEGCubeTexture` as class name, but since the canvas supports more than just JPEGs I ended up with `ImageCubeTexture`. The only other name I can come up with is `CanvasCubeTexture`, but I don't know if that more appropriate.

Reply:
Not a fan of EquirectangularCubeTexture?

Reply:
Refactored everything for `EquirectangularCubeTexture`.